osu.Game/Storyboards/Drawables/DrawableStoryboardSample.cs

```diff
@@ -28,12 +34,18 @@ public DrawableStoryboardSample(StoryboardSampleInfo sampleInfo)
         }
 
         [BackgroundDependencyLoader]
-        private void load(IBindable<WorkingBeatmap> beatmap)
+        private void load(IBindable<WorkingBeatmap> beatmap, IBindable<IReadOnlyList<Mod>> mods, Player player)
         {
-            channel = beatmap.Value.Skin.GetSample(sampleInfo);
+            Channel = beatmap.Value.Skin.GetSample(sampleInfo);
+            if (Channel == null)
+                return;
+
+            Channel.Volume.Value = sampleInfo.Volume / 100.0;
+
+            foreach (var mod in mods.Value.OfType<IApplicableToSample>())
+                mod.ApplyToSample(Channel);
 
-            if (channel != null)
-                channel.Volume.Value = sampleInfo.Volume / 100.0;
+            Channel.AddAdjustment(AdjustableProperty.Frequency, player.GameplayClockContainer.UserPlaybackRate);
```

Comment:
Please clarify what this is fixing - I assume it's fixing the storyboard samples not adhering to the playback rate set in the replay settings?

I'm not a fan of this solution as the rate shouldn't be exposed like that. I would have preferred the current playback rate to be stored in `GameplayBeatmap` which could be DI'd in here without having to make it public in the gameplay clock container.

Reply:
My bad, It was intended to fix playback rate in replay settings. I'm reverting these changes and will rework those specific changes for a new PR. Thanks for the feedback.